nrrd/tests/test_reading.py

```diff
@@ -65,9 +62,23 @@ def test_read_detached_header_and_data(self):
 
         np.testing.assert_equal(self.expected_header, header)
         np.testing.assert_equal(data, self.expected_data)
-
-        # Test that the data read is able to be edited
-        self.assertTrue(data.flags['WRITEABLE'])
+        
+    def test_read_detached_header_and_data_with_byteskip_minus1(self):
+        expected_header = self.expected_header
+        expected_header[u'data file'] = os.path.basename(RAW_DATA_FILE_PATH)
+        expected_header[u'byte skip'] = -1
+
+        data, header = nrrd.read(RAW_BYTESKIP_NHDR_FILE_PATH)
+        
+        print('Given')
+        print(expected_header)
+        
+        print('Found')
+        print(header)
+        
+        np.testing.assert_equal(self.expected_header, header)
+        np.testing.assert_equal(data, self.expected_data)       
```

Comment:
I'm not entirely sure how this tests the behavior of the byte_skip = -1. Since it is just seeking backwards to the beginning of the file.

I would change the data_file and add some garbage to the front of it. That way seeking from the front won't work but seeking backwards will! Then have one test using byte_skip = -1, and one test that should fail using no byte_skip.

Reply:
With this in mind, don't be afraid to add a new NRRD file to test with. Just use some fake/synthesized data that works for your purpose.